src/backend/lib/utils.js

```diff
@@ -64,16 +111,42 @@ function installerCompare(a, b) {
 function isUpgradeCandidate({ deviceInfo, installer, currentApkInfo }) {
   // Filter out the current APK
   if (installer.hash === currentApkInfo.hash) return false;
+  // Check for 0 or less APK size
+  if (installer.size <= 0) return false;
   // TODO: 64 bit devices can run 32 bit APKs
   const isSupportedAbi = !!installer.arch.find(arch =>
     deviceInfo.supportedAbis.includes(arch)
   );
   if (!isSupportedAbi) return false;
+  log(
+    "installer.minSdkVersion > deviceInfo.sdkVersion",
+    installer.minSdkVersion > deviceInfo.sdkVersion
+  );
   if (installer.minSdkVersion > deviceInfo.sdkVersion) return false;
```

Comment:
This is going to make the debug logs super noisy (and harder to parse in my opinion) because this function is called multiple times. I suggest we only log the reason an installer is rejected, e.g. only log just before returning `false` like:

```suggestion
  if (installer.minSdkVersion > deviceInfo.sdkVersion) {
  	log(`installer rejected: minSdkVersion ${installer.minSdkVersion} is not supported by this device`)
    return false;
  }
```